packages/dd-trace/src/appsec/blocking.js

```diff
@@ -127,11 +132,33 @@ function block (req, res, rootSpan, abortController, actionParameters = defaultB
     rootSpan?.setTag('_dd.appsec.block.failed', 1)
     log.error('[ASM] Blocking error', err)
 
+    // TODO: if blocking fails, then the response will never be sent ?
+
     updateBlockFailureMetric(req)
     return false
   }
 }
 
+function delegateBlock (req, res) {
+  const args = arguments
+
+  return new Promise((resolve) => {
+    // ignore subsequent delegations by never calling their resolve()
+    if (blockDelegations.has(res)) return
+
+    blockDelegations.set(res, { args, resolve })
+  })
+}
+
+function blockDelegate (res) {
+  const delegation = blockDelegations.get(res)
+  if (delegation) {
+    const result = block.apply(this, delegation.args)
+    delegation.resolve(result)
+    return result
+  }
+}
+
```

Comment:
Nit: Regarding the functions naming here, both names are quite similar (they are partial anagrams) and `blockDelegate` might benefit from a clearer verb. Myabe something like `registerDelegationBlock` would make the intent of the function a bit more explicit?

Not a blocker, but I think renaming could improve readability here.

Reply:
`registerBlockDelegation` is better english, wdyt ?

Reply:
Your proposal sounds better 👍

Reply:
like that ? [`fd62980` (#6081)](https://github.com/DataDog/dd-trace-js/pull/6081/commits/fd62980c62a4c81c13ef55e5941103c6dc704bbc)